### PR TITLE
Update `urllib3.fields.py` type hints

### PR DESCRIPTION
**Pull request by kkirsche**
This is required to address some re-exports in https://github.com/python/typeshed/pull/8439

The source code is available here:
https://github.com/urllib3/urllib3/blob/main/src/urllib3/fields.py

### CODE REVIEW

File: stubs/urllib3/urllib3/fields.pyi [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+        headers: Mapping[str, str] | None = ...,
+        header_formatter: Callable[[str, _FieldValue], str] | None = ...,
+    ) -> None: ...
     @classmethod
-    def from_tuples(cls, fieldname, value): ...
-    def render_headers(self): ...
-    def make_multipart(self, content_disposition=..., content_type=..., content_location=...): ...
+    def from_tuples(
+        cls, fieldname: str, value: _FieldValueTuple, header_formatter: Callable[[str, _FieldValue], str] | None = ...
+    ) -> "RequestField": ...
```
Comment (AlexWaygood) [inline stubs/urllib3/urllib3/fields.pyi:27]:
```suggestion
    ) -> RequestField: ...
```

File: stubs/urllib3/urllib3/fields.pyi [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
 
-def guess_content_type(filename, default=...): ...
-def format_header_param(name, value): ...
+_FieldValue: TypeAlias = str | bytes
+_FieldValueTuple: TypeAlias = Union[_FieldValue, tuple[str, _FieldValue], tuple[str, _FieldValue, str]]
+
+def guess_content_type(filename: str | None, default: str = ...) -> str: ...
+def format_header_param_rfc2231(name: str, value: _FieldValue) -> str: ...
+def format_header_param_html5(name: str, value: _FieldValue) -> str: ...
+def format_header_param(name: str, value: _FieldValue) -> str: ...
```
Comment (srittau) [inline stubs/urllib3/urllib3/fields.pyi:11]: To match the implemenation:

```suggestion
format_header_param = format_header_param_html5
```

File: stubs/urllib3/urllib3/fields.pyi [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
     data: Any
     headers: Any
-    def __init__(self, name, data, filename=..., headers=...) -> None: ...
+    def __init__(
+        self,
+        name: str,
+        data: _FieldValue,
+        filename: str | None = ...,
+        headers: Mapping[str, str] | None = ...,
+        header_formatter: Callable[[str, _FieldValue], str] | None = ...,
```
Comment (srittau) [inline stubs/urllib3/urllib3/fields.pyi:22]: From my reading the code, passing `None` would be an error:

```suggestion
        header_formatter: Callable[[str, _FieldValue], str] = ...,
```

File: stubs/urllib3/urllib3/fields.pyi [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
+    ) -> None: ...
     @classmethod
-    def from_tuples(cls, fieldname, value): ...
-    def render_headers(self): ...
-    def make_multipart(self, content_disposition=..., content_type=..., content_location=...): ...
+    def from_tuples(
+        cls, fieldname: str, value: _FieldValueTuple, header_formatter: Callable[[str, _FieldValue], str] | None = ...
```
Comment (srittau) [inline stubs/urllib3/urllib3/fields.pyi:26]: Same here:

```suggestion
        cls, fieldname: str, value: _FieldValueTuple, header_formatter: Callable[[str, _FieldValue], str] = ...
```

File: stubs/urllib3/urllib3/fields.pyi (hunk 1 above)
Comment (srittau) [inline stubs/urllib3/urllib3/fields.pyi]:
```suggestion
    ) -> RequestField: ...
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! Notes below.

File: stubs/urllib3/urllib3/fields.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stubs/urllib3/urllib3/fields.pyi]: I already fixed this one :)

File: stubs/urllib3/urllib3/fields.pyi (hunk 2 above)
Comment (kkirsche) [reply stubs/urllib3/urllib3/fields.pyi:11]: Good catch. I had read 2.x.x code when it was supposed to be https://github.com/urllib3/urllib3/blob/1.26.x/src/urllib3/fields.py where it is required

**Comment by kkirsche** [comment]
Thank you for the review! I believe I've addressed it all :)

**Review by srittau — APPROVED** [review]

**Pull request merged**